autopush/router/simple.py

```diff
@@ -188,21 +194,42 @@ def _send_notification(self, uaid, node_id, notification):
                               "version": notification.version,
                               "data": notification.data})
         url = node_id + "/push/" + uaid
-        d = self.ap_settings.agent.request(
+        request = self.ap_settings.agent.request(
             "PUT",
             url.encode("utf8"),
             bodyProducer=FileBodyProducer(StringIO(payload)),
         )
-        d.addCallback(IgnoreBody.ignore)
-        return d
+        # Only time out actual calls, not mocks.
+        if isinstance(request, defer.Deferred):
+            timeout = reactor.callLater(self.ap_settings.router_timeout,
+                                        request.cancel)
```

Comment:
I meant we'd drop the value of the connectTimeout substantially, ie maybe 500ms, this means if it ends up making two dead connect calls (our worst-case node situation in routing), our max time taken is 1sec, not 10sec. Sorry, wasn't clear enough what I meant to 'drop'.

I'm wary of canceling the entire routing call because the logic in it wasn't made with the assumption that we'd nuke the whole run-through externally. It only was created with the assumption that talking to nodes sometimes might fail.